vfs-azure/src/main/java/com/sludev/commons/vfs2/provider/azure/AzFileNameParser.java

```diff
@@ -16,33 +16,82 @@
  */
 package com.sludev.commons.vfs2.provider.azure;
 
+import org.apache.commons.vfs2.FileName;
+import org.apache.commons.vfs2.FileSystemException;
+import org.apache.commons.vfs2.FileType;
 import org.apache.commons.vfs2.provider.FileNameParser;
-import org.apache.commons.vfs2.provider.URLFileNameParser;
+import org.apache.commons.vfs2.provider.UriParser;
+import org.apache.commons.vfs2.provider.VfsComponentContext;
+import org.apache.commons.vfs2.provider.url.UrlFileNameParser;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+
 /**
  * Used for defining / parsing a provided FileName object.
- * 
+ *
  * This name should adhere to a URL structure, complete with an 'authority'
- * 
+ *
  * &lt;scheme&gt;://&lt;host_or_authority&gt;/&lt;container&gt;/&lt;file_path&gt;
  * E.g. azsb://myAccount.blob.core.windows.net/myContainer/path/to/file.txt
- * 
+ *
  */
-public class AzFileNameParser extends URLFileNameParser
-{
+public class AzFileNameParser extends UrlFileNameParser {
+
     private static final Logger log = LoggerFactory.getLogger(AzFileNameParser.class);
-    
+
     private static final AzFileNameParser INSTANCE = new AzFileNameParser();
 
-    public AzFileNameParser()
-    {
-        super(80);
+
+    private AzFileNameParser() {
     }
 
     public static FileNameParser getInstance()
     {
         return INSTANCE;
     }
+
+
+    @Override
+    public FileName parseUri(final VfsComponentContext context, final FileName base, final String uri)
+            throws FileSystemException {
+
+        // azbs://flexqaooflexstorage.blob.core.windows.net/flex-azure-vfs-test/test01.tmp
+
+        StringBuilder sb = new StringBuilder(uri);
+
+        UriParser.normalisePath(sb);
+
+        String normalizedUri = sb.toString();
+
+        String scheme = normalizedUri.substring(0, normalizedUri.indexOf(':'));
+        String path = normalizedUri.substring(6);
```

Comment:
should this not be substring after the `:` separator, rather than the magic number 6?